includes/object-cache.php

```diff
@@ -636,6 +640,116 @@ protected function connectUsingPredis( $parameters ) {
+            $to_load[] = 'Cluster.php';
+
+            if ( true
+                && defined( 'WP_REDIS_SENTINEL' )
+                && ! class_exists( 'Credis_Sentinel' )
+            ) {
+                $to_load[] = 'Sentinel.php';
+            }
+        }
+
+        foreach ( $to_load as $sub_path ) {
+            $path = $creds_path . $sub_path;
+            if ( file_exists( $path ) ) {
+                require_once $path;
+            } else {
+                throw new Exception(
+                    sprintf(
+                        'Credis "%s" library not found. Re-install Redis Cache plugin or delete object-cache.php.',
+                        str_replace( '.php', '', $sub_path )
+                    )
+                );
+            }
+        }
+
+        if ( defined( 'WP_REDIS_SHARDS' ) ) {
+            // TODO: untested!
+            // Sharding not supported?
+            throw new Exception( 'Sharding not supported by bundled credis library. Please review your Redis Cache configuration.' );
+        } elseif ( false
+            || defined( 'WP_REDIS_CLUSTER' )
+            || defined( 'WP_REDIS_SERVERS' )
+        ) {
+            // TODO: untested!
+            $parameters['db'] = $parameters['database'];
+            
+            $is_cluster = defined( 'WP_REDIS_CLUSTER' );
+            $clients = $is_cluster ? WP_REDIS_CLUSTER : WP_REDIS_SERVERS;
+
+            foreach ( $clients as &$client ) {
+                $add_params['host'] = $client;
+                if ( $is_cluster ) {
+                    $add_params['master'] = false !== stripos( $client, 'alias=master' );
+                }
+                $client = array_merge( $parameters, $add_params );
+            }
+
+            $this->redis = new Credis_Cluster( $clients );
+
+            // Supply diagnostic information.
+            $args = $clients;
+            $args[ $is_cluster ? 'is_cluster' : 'is_multi'] = true;
+
+        } else {
+            $host = 'unix' === $parameters['scheme'] ? $parameters['path'] : $parameters['host'];
+
+            $args = [
+                "{$parameters['scheme']}://{$host}",
+                $parameters['port'],
+                $parameters['timeout'],
+                '',
+                isset( $parameters['database'] ) ? $parameters['database'] : 0,
+                isset( $parameters['password'] ) ? $parameters['password'] : null,
+            ];
+            $this->redis = new Credis_Client( ...$args );
+        }
+
+        if ( defined( 'WP_REDIS_SENTINEL' ) ) {
+            // TODO: untested!
+            $this->redis = new Credis_Sentinel( $this->redis );
+            $args['is_sentinel'] = true;
+        }
+
+        // Credis uses phpredis if it detects it unless we force it to run standalone.
+        $this->redis->forceStandalone();
+
+        $this->redis->connect();
+
+        $this->diagnostics = array_merge(
+            [ 'client' => $client ], // Sadly Credis provides no version information.
```

Comment:
If we use fixed versions in the `composer.json` we can parse it and add the version here 👍🏻

Reply:
Thought about that but it seems like a quite hacky approach. Other than opening an issue on Credits itself I don't see a better one.

Reply:
Let's do that, I'll open an issue and see how fast they are.

`Latest commit on Nov 26, 2019` yikes...

Reply:
Waiting for merge: https://github.com/colinmollenhour/credis/pull/136

Reply:
Merged: https://github.com/colinmollenhour/credis/pull/136

Reply:
Resolved in ff16ed915310e05fb0908cc6e50b43a29d410b7f